### PR TITLE
Adds delete functionality for a course

### PR DESCRIPTION
**Pull request by ghatighorias**

### CODE REVIEW

File: web/controllers/class_controller.ex [hunk 1]
```diff
@@ -51,12 +51,12 @@ defmodule CoursePlanner.ClassController do
...
-#    class = Repo.get!(Class, id)
-#    ClassHelper.delete(class)
-
-#    conn
-#    |> put_flash(:info, "Class deleted successfully.")
-#    |> redirect(to: class_path(conn, :index))
-#  end
+  def delete(conn, %{"id" => id}) do
+    class = Repo.get!(Class, id)
+    ClassHelper.delete(class)
```
Comment (rhnonose) [inline web/controllers/class_controller.ex]: Can you check the return for successful delete?

File: test/controllers/class_controller_test.exs [hunk 2]
```diff
@@ -221,10 +221,52 @@ defmodule CoursePlanner.ClassControllerTest do
...
+  test "hard deletes chosen resource when status is Planned", %{conn: conn} do
+    {:ok, created_course} = create_course()
+    class_args = %Class{course_id: created_course.id, date: Ecto.Date.from_erl({2010, 01, 01}), starting_at: Ecto.Time.from_erl({13, 0, 0}), finishes_at: Ecto.Time.from_erl({14, 0, 0}), status: "Planned"}
+    class = Repo.insert! class_args
+    conn = delete conn, class_path(conn, :delete, class)
+    assert redirected_to(conn) == class_path(conn, :index)
+    refute Repo.get(Class, class.id)
+  end
+
+  test "soft deletes chosen resource when status is Active", %{conn: conn} do
```
Comment (rhnonose) [inline test/controllers/class_controller_test.exs]: I don't think it's necessary to test all status for the delete action.
If you wish so, try to make a generic code so it's not too verbose, like:
```elixir
  Enum.map ["Active", "Frozen"], fn(status) ->
    test "soft deletes chosen resource when status is #{status}", %{conn: conn} do
      #your test here
    end
  end
```

File: web/controllers/class_controller.ex [hunk 3]
```diff
@@ -51,12 +51,12 @@ defmodule CoursePlanner.ClassController do
...
-#  def delete(conn, %{"id" => id}) do
-#    class = Repo.get!(Class, id)
-#    ClassHelper.delete(class)
-
-#    conn
-#    |> put_flash(:info, "Class deleted successfully.")
-#    |> redirect(to: class_path(conn, :index))
-#  end
+  def delete(conn, %{"id" => id}) do
+    class = Repo.get!(Class, id)
```
Comment (rhnonose) [inline web/controllers/class_controller.ex]: You can put the get inside the `ClassHelper` module to remove the `Repo` dependency from the controller.

File: test/controllers/class_controller_test.exs [hunk 4]
```diff
@@ -221,10 +221,30 @@ defmodule CoursePlanner.ClassControllerTest do
     assert html_response(conn, 200) =~ "Edit class"
```
Comment (ggpasqualino) [inline test/controllers/class_controller_test.exs]: could you add a test for deleting a non-existent id?

Comment (ghatighorias) [inline test/controllers/class_controller_test.exs]: done

**Pull request merged**